src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseSpanClearInsteadOfFill.cs

```diff
@@ -0,0 +1,155 @@
+    public sealed class UseSpanClearInsteadOfFillAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "CA1849";
+        internal const string FillMethod = "Fill";
+        internal const string ClearMethod = "Clear";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UseSpanClearInsteadOfFillTitle), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UseSpanClearInsteadOfFillMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UseSpanClearInsteadOfFillDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static readonly DiagnosticDescriptor s_Rule = DiagnosticDescriptorHelper.Create(
+            DiagnosticId,
+            s_localizableTitle,
+            s_localizableMessage,
+            DiagnosticCategory.Performance,
+            RuleLevel.BuildWarning,
+            description: s_localizableDescription,
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(s_Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var compilation = context.Compilation;
+            var spanType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemSpan1);
+
+            if (spanType == null)
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(
+                operationContext =>
+                {
+                    var invocation = (IInvocationOperation)operationContext.Operation;
+
+                    if (!SymbolEqualityComparer.Default.Equals((invocation.Instance.Type as INamedTypeSymbol)?.ConstructedFrom, spanType))
+                    {
+                        return;
+                    }
+
+                    if (invocation.TargetMethod.Name != FillMethod)
+                    {
+                        return;
+                    }
+
+                    if (invocation.Arguments.Length != 1)
+                    {
+                        return;
+                    }
```

Comment:
I'm not sure. But if these checks are cheaper than symbol comparison, consider moving them up.

Reply:
Actually, I'd try getting the method symbol at compilation start, then do symbol comparison against `invocation.TargetMethod`.